Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -2285,6 +2285,71 @@ void Document::set_cookie(StringView cookie_string, Cookie::Source source)
     page().client().page_did_set_cookie(m_url, cookie.value(), source);
 }
 
+String Document::fg_color() const
+{
+    if (auto* body_element = body(); body_element && !is<HTML::HTMLFrameSetElement>(*body_element))
+        return body_element->get_attribute_value(HTML::AttributeNames::text);
+    return ""_string;
+}
+
+void Document::set_fg_color(String const& value)
+{
+    if (auto* body_element = body(); body_element && !is<HTML::HTMLFrameSetElement>(*body_element))
+        MUST(body_element->set_attribute(HTML::AttributeNames::text, value));
```

Comment:
`TRY()` should be used here (and in the other setters), so that any error returned by `set_attribute()` is propagared to the caller instead of the program crashing. To do this you need to change the setters' return type to `WebIDL::ExceptionOr<void>`.

Sorry for the back and forth, I was sure I mentioned this in my last review :sweat_smile:

Reply:
I believe it's okay to use `MUST()` here as the only possible exception from `set_attribute()` is when you're passing an invalid attribute name.

Reply:
> I believe it's okay to use `MUST()` here as the only possible exception from `set_attribute()` is when you're passing an invalid attribute name.

You're absolutely right.

Ignore me, what's there currently is fine!